pyk/src/pyk/tests/test_match.py

```diff
@@ -30,10 +35,13 @@ def test_match_and_subst(self):
             with self.subTest(i=i):
                 # When
                 subst = match(pattern, term)
+                subst_cterm = _as_cterm(pattern).match(_as_cterm(term))
 
                 # Then
                 self.assertIsNotNone(subst)
                 self.assertEqual(subst(pattern), term)
+                self.assertIsNotNone(subst_cterm)
+                self.assertEqual(subst_cterm(pattern), term)
```

Comment:
I think `CTerm` should have its own set of tests.

Reply:
Done.